airflow/providers_manager.py

```diff
@@ -950,6 +957,14 @@ def _discover_auth_backends(self) -> None:
                     if _sanity_check(provider_package, auth_backend_module_name + ".init_app", provider):
                         self._api_auth_backend_module_names.add(auth_backend_module_name)
 
+    def _discover_executors(self) -> None:
+        """Retrieve all executors defined in the providers."""
+        for provider_package, provider in self._provider_dict.items():
+            if provider.data.get("executors"):
+                for executors_class_name in provider.data["executors"]:
+                    if _sanity_check(provider_package, executors_class_name, provider):
```

Comment:
Out of scope, but we should likely rename that method at some point.

Reply:
Yes. NOW.